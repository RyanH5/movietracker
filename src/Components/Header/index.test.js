import React from 'react';
import { shallow } from 'enzyme';
import {Header} from './index';

// jest.mock('./index.js');

describe('Header', () => {
  let wrapper;
  let mockUserLogout;
  let mockUser;

  beforeEach(() => {
    mockUserLogout = jest.fn();
    mockUser = {name: 'Bill', id: 4, loginStatus: true}
    wrapper = shallow(<Header userLogout={mockUserLogout}/>);
  });

  it('should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should call userLogout on click', ()=> {
    wrapper.find('.logout').last().simulate('click');

    expect(mockUserLogout).toHaveBeenCalledWith(mockUser)
  });
});