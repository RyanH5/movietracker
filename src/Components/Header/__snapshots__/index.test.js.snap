// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    userLogout={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="App-title"
>
          Movie Tracker
</h1>,
        <div
          className="logged-out-nav"
>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="nav home"
                    to="/"
          >
                    Home
          </NavLink>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="nav"
                    to="/login"
          >
                    Login
                    
                    SignUp
          </NavLink>
</div>,
      ],
      "className": "header-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Movie Tracker",
          "className": "App-title",
        },
        "ref": null,
        "rendered": "Movie Tracker",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav home"
              to="/"
>
              Home
</NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav"
              to="/login"
>
              Login
              <br />
              SignUp
</NavLink>,
          ],
          "className": "logged-out-nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Home",
              "className": "nav home",
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": Array [
                "Login",
                <br />,
                "SignUp",
              ],
              "className": "nav",
              "to": "/login",
            },
            "ref": null,
            "rendered": Array [
              "Login",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              "SignUp",
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="App-title"
>
            Movie Tracker
</h1>,
          <div
            className="logged-out-nav"
>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="nav home"
                        to="/"
            >
                        Home
            </NavLink>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="nav"
                        to="/login"
            >
                        Login
                        <br />
                        SignUp
            </NavLink>
</div>,
        ],
        "className": "header-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Movie Tracker",
            "className": "App-title",
          },
          "ref": null,
          "rendered": "Movie Tracker",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav home"
                to="/"
>
                Home
</NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav"
                to="/login"
>
                Login
                <br />
                SignUp
</NavLink>,
            ],
            "className": "logged-out-nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Home",
                "className": "nav home",
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": Array [
                  "Login",
                  <br />,
                  "SignUp",
                ],
                "className": "nav",
                "to": "/login",
              },
              "ref": null,
              "rendered": Array [
                "Login",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                "SignUp",
              ],
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
